invenio_app_ils/circulation/indexer.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Loan indexer APIs."""
+
+from datetime import datetime
+
+from celery import shared_task
+from flask import current_app
+from invenio_circulation.pidstore.pids import CIRCULATION_LOAN_PID_TYPE
+from invenio_indexer.api import RecordIndexer
+
+from invenio_app_ils.indexer import ReferencedRecordsIndexer
+from invenio_app_ils.pidstore.pids import DOCUMENT_PID_TYPE, ITEM_PID_TYPE
+from invenio_app_ils.proxies import current_app_ils
+
+
+@shared_task(ignore_result=True)
+def index_referenced_records(loan):
+    """Index referenced records."""
+    indexer = ReferencedRecordsIndexer()
+    indexed = dict(pid_type=CIRCULATION_LOAN_PID_TYPE, record=loan)
+    referenced = []
+
+    # fetch and index the document
+    document_cls = current_app_ils.document_record_cls
+    document = document_cls.get_record_by_pid(loan["document_pid"])
+    referenced.append(dict(pid_type=DOCUMENT_PID_TYPE, record=document))
+
+    # fetch and index the item
+    if loan.get("item_pid"):
+        item_cls = current_app_ils.item_record_cls
+        item = item_cls.get_record_by_pid(loan["item_pid"])
+        referenced.append(dict(pid_type=ITEM_PID_TYPE, record=item))
+
+    # index the document
+    indexer.index(indexed, referenced)
+
+
+class LoanIndexer(RecordIndexer):
+    """Indexer class for Loan record."""
+
+    def index(self, loan, arguments=None, **kwargs):
+        """Index an Loan."""
+        super().index(loan)
+        eta = datetime.utcnow() + current_app.config["ILS_INDEXER_TASK_DELAY"]
```

Comment:
nit: are we suppose to use arrow instead of datetime?

Reply:
good question! :) No I don't think so, this is just to delay the execution of the celery task, it is like `timedelta(2 secs)`